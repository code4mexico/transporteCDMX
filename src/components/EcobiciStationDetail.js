import React, { memo } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { Button, Title } from 'react-native-paper'
import EcobiciStationModel from '../models/EcobiciStation'
import theme, { sharedStyles } from '../styles'
import { translate } from '../i18n'

const EcobiciStationDetail = ({ station, fetchEcobiciStations, isLoading }) => {
  return (
    <View style={sharedStyles.p3}>
      <Title style={sharedStyles.textAlignCenter}>{station.status}</Title>
      <Title style={[sharedStyles.textAlignCenter, theme.colors.colorSecondary, sharedStyles.mt3]}>
        {station.stationType}
      </Title>
      <View style={[sharedStyles.flexDRow, sharedStyles.mt3]}>
        <View
          style={[sharedStyles.flex1, sharedStyles.flexDRow, sharedStyles.justifyContentCenter]}>
          <Title>{`${translate('bikes')}: ${station.availability?.bikes}`}</Title>
        </View>
        <View
          style={[sharedStyles.flex1, sharedStyles.flexDRow, sharedStyles.justifyContentCenter]}>
          <Title>{`${translate('slots')}: ${station.availability?.slots}`}</Title>
        </View>
      </View>
      <Button
        theme={theme}
        mode="text"
        style={sharedStyles.mt3}
        labelStyle={theme.colors.colorSecondary}
        onPress={fetchEcobiciStations}>
        {isLoading ? translate('loading_stations') : translate('reload_information')}
      </Button>
    </View>
  )
}

EcobiciStationDetail.propTypes = {
  isLoading: PropTypes.bool,
  fetchEcobiciStations: PropTypes.func.isRequired,
  station: PropTypes.instanceOf(EcobiciStationModel),
}

export default memo(EcobiciStationDetail)
