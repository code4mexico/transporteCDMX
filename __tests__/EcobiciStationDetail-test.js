import React from 'react'
import { shallow } from 'enzyme'
import EcobiciStationDetail from '../src/components/EcobiciStationDetail'
import ecobiciStationFixture from './fixtures/ecobiciStation-fixture'

const fetchEcobiciStations = jest.fn

describe('EcobiciStationDetail', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(
      <EcobiciStationDetail
        station={ecobiciStationFixture}
        fetchEcobiciStations={fetchEcobiciStations}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('it renders correctly when loading', () => {
    const wrapper = shallow(
      <EcobiciStationDetail
        station={ecobiciStationFixture}
        fetchEcobiciStations={fetchEcobiciStations}
        isLoading
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
