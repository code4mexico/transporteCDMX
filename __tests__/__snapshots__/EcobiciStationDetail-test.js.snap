// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcobiciStationDetail it renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 17,
    }
  }
>
  <Title
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Estatus de la estación: Abierta
  </Title>
  <Title
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#2B45AC",
        },
        Object {
          "marginTop": 17,
        },
      ]
    }
  >
    Estacion de bicicletas
  </Title>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginTop": 17,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Title>
        [missing "en.bikes" translation]: 3
      </Title>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Title>
        [missing "en.slots" translation]: 24
      </Title>
    </View>
  </View>
  <withTheme(Button)
    labelStyle={
      Object {
        "color": "#2B45AC",
      }
    }
    mode="text"
    onPress={[Function]}
    style={
      Object {
        "marginTop": 17,
      }
    }
    theme={
      Object {
        "animation": Object {
          "scale": 1,
        },
        "colors": Object {
          "accent": "#2B45AC",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#EDEDED",
          "bgBackground": Object {
            "backgroundColor": "#EDEDED",
          },
          "bgBlack": Object {
            "backgroundColor": "#000",
          },
          "bgBlackOpacity": Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          },
          "bgDanger": Object {
            "backgroundColor": "#FF4D4D",
          },
          "bgLightGray": Object {
            "backgroundColor": "#E5E5E5",
          },
          "bgPrimary": Object {
            "backgroundColor": "#FFF",
          },
          "bgPrimaryOpacity": Object {
            "backgroundColor": "rgba(255,255,255,0.5)",
          },
          "bgSecondary": Object {
            "backgroundColor": "#2B45AC",
          },
          "bgWhite": Object {
            "backgroundColor": "#FFF",
          },
          "black": "#000",
          "blackOpacity": "rgba(0,0,0,0.5)",
          "colorDanger": Object {
            "color": "#FF4D4D",
          },
          "colorMediumGray": Object {
            "color": "#A1A1A1",
          },
          "colorPrimary": Object {
            "color": "#FFF",
          },
          "colorPrimaryOpacity": Object {
            "color": "rgba(255,255,255,0.5)",
          },
          "colorSecondary": Object {
            "color": "#2B45AC",
          },
          "colorWhite": Object {
            "color": "#FFF",
          },
          "danger": "#FF4D4D",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "lightGray": "#E5E5E5",
          "mediumGray": "#A1A1A1",
          "notification": "#f50057",
          "onBackground": "#000000",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#FFF",
          "primaryOpacity": "rgba(255,255,255,0.5)",
          "secondary": "#2B45AC",
          "success": "#4e9f41",
          "surface": "#ffffff",
          "text": "#000000",
          "warning": "#FFCA45",
          "white": "#FFF",
        },
        "dark": false,
        "fonts": Object {
          "light": Object {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": Object {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": Object {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": Object {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 2,
      }
    }
  >
    [missing "en.reload_information" translation]
  </withTheme(Button)>
</View>
`;

exports[`EcobiciStationDetail it renders correctly when loading 1`] = `
<View
  style={
    Object {
      "padding": 17,
    }
  }
>
  <Title
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Estatus de la estación: Abierta
  </Title>
  <Title
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#2B45AC",
        },
        Object {
          "marginTop": 17,
        },
      ]
    }
  >
    Estacion de bicicletas
  </Title>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginTop": 17,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Title>
        [missing "en.bikes" translation]: 3
      </Title>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Title>
        [missing "en.slots" translation]: 24
      </Title>
    </View>
  </View>
  <withTheme(Button)
    labelStyle={
      Object {
        "color": "#2B45AC",
      }
    }
    mode="text"
    onPress={[Function]}
    style={
      Object {
        "marginTop": 17,
      }
    }
    theme={
      Object {
        "animation": Object {
          "scale": 1,
        },
        "colors": Object {
          "accent": "#2B45AC",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#EDEDED",
          "bgBackground": Object {
            "backgroundColor": "#EDEDED",
          },
          "bgBlack": Object {
            "backgroundColor": "#000",
          },
          "bgBlackOpacity": Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          },
          "bgDanger": Object {
            "backgroundColor": "#FF4D4D",
          },
          "bgLightGray": Object {
            "backgroundColor": "#E5E5E5",
          },
          "bgPrimary": Object {
            "backgroundColor": "#FFF",
          },
          "bgPrimaryOpacity": Object {
            "backgroundColor": "rgba(255,255,255,0.5)",
          },
          "bgSecondary": Object {
            "backgroundColor": "#2B45AC",
          },
          "bgWhite": Object {
            "backgroundColor": "#FFF",
          },
          "black": "#000",
          "blackOpacity": "rgba(0,0,0,0.5)",
          "colorDanger": Object {
            "color": "#FF4D4D",
          },
          "colorMediumGray": Object {
            "color": "#A1A1A1",
          },
          "colorPrimary": Object {
            "color": "#FFF",
          },
          "colorPrimaryOpacity": Object {
            "color": "rgba(255,255,255,0.5)",
          },
          "colorSecondary": Object {
            "color": "#2B45AC",
          },
          "colorWhite": Object {
            "color": "#FFF",
          },
          "danger": "#FF4D4D",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "lightGray": "#E5E5E5",
          "mediumGray": "#A1A1A1",
          "notification": "#f50057",
          "onBackground": "#000000",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#FFF",
          "primaryOpacity": "rgba(255,255,255,0.5)",
          "secondary": "#2B45AC",
          "success": "#4e9f41",
          "surface": "#ffffff",
          "text": "#000000",
          "warning": "#FFCA45",
          "white": "#FFF",
        },
        "dark": false,
        "fonts": Object {
          "light": Object {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": Object {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": Object {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": Object {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 2,
      }
    }
  >
    [missing "en.loading_stations" translation]
  </withTheme(Button)>
</View>
`;
